components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -116,6 +116,51 @@ const char kHideSelectorsInjectScript[] =
           };
         })();)";
 
+const char kRemovalsInjectScript[] =
+    R"((function() {
+          const selectorsToRemove = %s;
+          const classesToRemoveBySelector = %s;
+          const attributesToRemoveBySelector = %s;
+          const onMutation = (mutationList) => {
+            if (mutationList.find(
+                mutation => (
+                  mutation.addedNodes.length !== 0 && mutation.addedNodes.values().find(node => node.nodeType == 1)
+                )
+              )
+            ) {
+              execute();
+            }
+          };
+          const execute = () => {
+              selectorsToRemove.forEach(
+                selector => document.querySelectorAll(selector).forEach(elem => elem.remove())
+              );
+              for (const selector in classesToRemoveBySelector) {
```

Comment:
What is the avg/max size of classesToRemoveBySelector and the other lists here?

Currently we have a complexity about number_of_mutations * total_size_of_the_lists * querySelectorAll_complexity(iterating though all the nodes)

I suspect the current approach can literally hang some pages with a lot of mutations.

Reply:
What can I do to prevent this? I have changed it to use `.matches` rather than `querySelectorAll` but the complexity is still similar. I can't really reason about how large the lists are, but I don't think they will be too large (difficult to know though).

Reply:
@atuchin-m it shouldn't be many; usually 0 and maybe on the order of 10 in the worst cases.